presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -216,6 +217,159 @@ else if (rescaleFactor > 0) {
         }
     }
 
+    public static Slice add(Slice left, Slice right)
+    {
+        Slice result = unscaledDecimal();
+        add(left, right, result);
+        return result;
+    }
+
+    public static void add(Slice left, Slice right, Slice result)
+    {
+        long overflow = addReturnOverflow(left, right, result);
+        if (overflow != 0) {
+            throwOverflowException();
+        }
+    }
+
+    /**
+     * Instead of throwing overflow exception, this function returns:
+     * 0 when there was no overflow
+     * +1 when there was overflow
+     * -1 when there was underflow
+     */
+    public static long addReturnOverflow(Slice left, Slice right, Slice result)
```

Comment:
I am not sure. What about other methods:
`subtract`, `multiply`, `divide`, `rescale`, etc. Basically all methods throw overflow exception if the value doesn't fit and this is generally the required semantic. Suffixing them all with `Exact` (in order to be consistent) doesn't seem right since we care about overflow most of the time and it's part of "normal" decimal arithmetics.

`addReturnOverflow` name while very verbose specifies exactly what the method does (it has only two non-test usages anyway)